### PR TITLE
feat: add -C --camelCaseTypes option

### PR DESCRIPTION
**Pull request by matt-forster**
This option adds the ability to camel case just the type names - which
gives a good mix between using JS Standard Camel Case and still
following the actual definitions of the database.

The issue with using camel case for both the types and the keys
is that we would have to provide a layer within the programs using the
types to convert back to the original form if the attributes are
different in JS than in the schema.

Issues with a mapping layer arise when a database has inconsistent naming conventions; 
we would have to provide some sort of mapping file instead of making assumptions
to achieve correct conversion.

The types on the other hand, only exist in JS and therefore can be named
whatever we want when generating the types.

### CODE REVIEW

File: bin/schemats-postgres.ts [hunk 1]
```diff
@@ -17,6 +17,7 @@ export const postgres = async (program: Command): Promise<void> => {
         .option('-t, --tables <tables...>', 'the tables within the schema')
         .option('-f, --typesFile <typesFile>', 'the file where jsonb types can be imported from')
         .option('-c, --camelCase', 'use camel case for enums and table names')
+        .option('-C, --camelCaseTypes', 'use camel case only for TS types - not modifying the table values')
```
Comment (matt-forster) [inline bin/schemats-postgres.ts:20]:
```suggestion
        .option('-C, --camelCaseTypes', 'use camel case only for TS types - not modifying the table attributes')
```

File: bin/schemats-postgres.ts [hunk 2]
```diff
@@ -17,6 +17,7 @@ export const postgres = async (program: Command): Promise<void> => {
         .option('-t, --tables <tables...>', 'the tables within the schema')
         .option('-f, --typesFile <typesFile>', 'the file where jsonb types can be imported from')
         .option('-c, --camelCase', 'use camel case for enums and table names')
```
Comment (matt-forster) [inline bin/schemats-postgres.ts:19]:
```suggestion
        .option('-c, --camelCase', 'use camel case for enums, table names, and column names')
```

File: bin/schemats-postgres.ts [hunk 3]
```diff
@@ -16,7 +16,8 @@ export const postgres = async (program: Command): Promise<void> => {
         .option('-s, --schema <schema>', 'the schema to use', 'public')
         .option('-t, --tables <tables...>', 'the tables within the schema')
         .option('-f, --typesFile <typesFile>', 'the file where jsonb types can be imported from')
-        .option('-c, --camelCase', 'use camel case for enums and table names')
+        .option('-c, --camelCase', 'use camel case for enums, table names, and column names')
+        .option('-C, --camelCaseTypes', 'use camel case only for TS types - not modifying the table attributes')
```
Comment (matt-forster) [inline bin/schemats-postgres.ts:20]:
```suggestion
        .option('-C, --camelCaseTypes', 'use camel case only for TS types - not modifying the column names')
```

File: bin/schemats-postgres.ts [hunk 4]
```diff
@@ -16,7 +16,8 @@ export const postgres = async (program: Command): Promise<void> => {
         .option('-s, --schema <schema>', 'the schema to use', 'public')
         .option('-t, --tables <tables...>', 'the tables within the schema')
         .option('-f, --typesFile <typesFile>', 'the file where jsonb types can be imported from')
-        .option('-c, --camelCase', 'use camel case for enums and table names')
+        .option('-c, --camelCase', 'use camel case for enums, table names, and column names')
+        .option('-C, --camelCaseTypes', 'use camel case only for TS types - not modifying the column names')
```
Comment (matt-forster) [inline bin/schemats-postgres.ts:20]:
```suggestion
        .option('-C, --camelCaseTypes', 'use camel case only for TS names - not modifying the column names')
```

**Comment by yasserf** [comment]
makes sense. I use a pg-node plugin that automatically converts everything it returns or is given between camelcase and snake case.

**Comment by yasserf** [comment]
will be doing a release with all three once i update changelog + documentation, latest by 2moro. Thanks for raising!

**Pull request merged**